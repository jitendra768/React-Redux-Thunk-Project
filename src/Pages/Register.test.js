/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import { fireEvent, render, screen } from "@testing-library/react";
import Register from "./Register";
import React from "react";

test("RegisterForm submits correct data", () => {
  const handleSubmit = jest.fn();
  const {asFragment}=render(<Register onSubmit={handleSubmit} />);

  fireEvent.change(screen.getByTestId("email-input"), {
    target: { value: "john@example.com" },
  });

  fireEvent.change(screen.getByTestId("password-input"), {
    target: { value: "password123" },
  });

  fireEvent.change(screen.getByTestId("confirmPassword-input"), {
    target: { value: "password123" },
  });

  fireEvent.click(screen.getByTestId("register-button"));
  expect(handleSubmit).toHaveBeenCalledWith({
    email: "john@example.com",
    password: "password123",
    confirmPassword: "password123",
  });
  expect(asFragment()).toMatchSnapshot()  //create snapshot
});
