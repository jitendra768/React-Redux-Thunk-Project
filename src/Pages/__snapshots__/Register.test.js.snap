// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm submits correct data 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col justify-center font-[sans-serif] sm:h-screen p-4"
  >
    <div
      class="max-w-md w-full mx-auto border border-gray-300 rounded-2xl p-8"
    >
      <div
        class="text-center mb-12"
      >
        <a
          href="javascript:void(0)"
        >
          <img
            alt="logo"
            class="w-40 inline-block"
            src="https://readymadeui.com/readymadeui.svg"
          />
        </a>
      </div>
      <form>
        <div
          class="space-y-6"
        >
          <div>
            <label
              class="text-gray-800 text-sm mb-2 block"
            >
              Email Id
            </label>
            <input
              class="text-gray-800 bg-white border border-gray-300 w-full text-sm px-4 py-3 rounded-md outline-blue-500"
              data-testid="email-input"
              name="email"
              placeholder="Enter email"
              type="text"
              value="john@example.com"
            />
          </div>
          <div>
            <label
              class="text-gray-800 text-sm mb-2 block"
            >
              Password
            </label>
            <input
              class="text-gray-800 bg-white border border-gray-300 w-full text-sm px-4 py-3 rounded-md outline-blue-500"
              data-testid="password-input"
              name="password"
              placeholder="Enter password"
              type="password"
              value="password123"
            />
          </div>
          <div>
            <label
              class="text-gray-800 text-sm mb-2 block"
            >
              Confirm Password
            </label>
            <input
              class="text-gray-800 bg-white border border-gray-300 w-full text-sm px-4 py-3 rounded-md outline-blue-500"
              data-testid="confirmPassword-input"
              name="confirmPassword"
              placeholder="Enter confirm password"
              type="password"
              value="password123"
            />
          </div>
          <div
            class="flex items-center"
          >
            <input
              class="h-4 w-4 shrink-0 text-blue-600 focus:ring-blue-500 border-gray-300 rounded"
              id="remember-me"
              name="remember-me"
              type="checkbox"
            />
            <label
              class="text-gray-800 ml-3 block text-sm"
              for="remember-me"
            >
              I accept the 
              <a
                class="text-blue-600 font-semibold hover:underline ml-1"
                href="javascript:void(0);"
              >
                Terms and Conditions
              </a>
            </label>
          </div>
        </div>
        <div
          class="!mt-8"
        >
          <button
            class="w-full py-3 px-4 text-sm tracking-wider font-semibold rounded-md text-white bg-blue-600 hover:bg-blue-700 focus:outline-none"
            data-testid="register-button"
            type="submit"
          >
            Create an account
          </button>
        </div>
        <p
          class="text-gray-800 text-sm mt-6 text-center"
        >
          Already have an account? 
          <a
            class="text-blue-600 font-semibold hover:underline ml-1"
            href="javascript:void(0);"
          >
            Login here
          </a>
        </p>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
